src/nv_ingest/schemas/chart_extractor_schema.py

```diff
@@ -0,0 +1,140 @@
+
+    Raises
+    ------
+    ValueError
+        If both gRPC and HTTP services are empty for any endpoint.
+
+    Config
+    ------
+    extra : str
+        Pydantic config option to forbid extra fields.
+    """
+
+    auth_token: Optional[str] = None
+
+    cached_endpoints: Tuple[Optional[str], Optional[str]] = (None, None)
+    cached_infer_protocol: str = ""
+
+    deplot_endpoints: Tuple[Optional[str], Optional[str]] = (None, None)
+    deplot_infer_protocol: str = ""
+
+    ## NOTE: Paddle isn't currently called independently of the cached NIM, but will be in the future.
+    paddle_endpoints: Tuple[Optional[str], Optional[str]] = (None, None)
+    paddle_infer_protocol: str = ""
+
+    @root_validator(pre=True)
+    def validate_endpoints(cls, values):
+        """
+        Validates the gRPC and HTTP services for all endpoints.
+
+        Ensures that at least one service (either gRPC or HTTP) is provided
+        for each endpoint in the configuration.
+
+        Parameters
+        ----------
+        values : dict
+            Dictionary containing the values of the attributes for the class.
+
+        Returns
+        -------
+        dict
+            The validated dictionary of values.
+
+        Raises
+        ------
+        ValueError
+            If both gRPC and HTTP services are empty for any endpoint.
+        """
+
+        def clean_service(service):
+            """Set service to None if it's an empty string or contains only spaces or quotes."""
+            if service is None or not service.strip() or service.strip(" \"'") == "":
+                return None
+            return service
+
+        for endpoint_name in ["cached_endpoints", "deplot_endpoints", "paddle_endpoints"]:
+            grpc_service, http_service = values.get(endpoint_name, (None, None))
+            grpc_service = clean_service(grpc_service)
+            http_service = clean_service(http_service)
+
+            if not grpc_service and not http_service:
+                raise ValueError(f"Both gRPC and HTTP services cannot be empty for {endpoint_name}.")
+
+            values[endpoint_name] = (grpc_service, http_service)
+
+        return values
+
+    class Config:
+        extra = "forbid"
+
+
+class ChartExtractorSchema(BaseModel):
+    """
+    Configuration schema for chart extraction processing settings.
+
+    Parameters
+    ----------
+    max_queue_size : int, default=1
+        The maximum number of items allowed in the processing queue.
+
+    n_workers : int, default=2
+        The number of worker threads to use for processing.
+
+    raise_on_failure : bool, default=False
+        A flag indicating whether to raise an exception if a failure occurs during chart extraction.
+
+    stage_config : Optional[ChartExtractorConfigSchema], default=None
+        Configuration for the chart extraction stage, including cached, deplot, and paddle service endpoints.
+    """
+
+    max_queue_size: int = 1
```

Comment:
Just curious is this is 1 because of an upstream limitation or just a random chosen value? If the later maybe we can adjust to something that makes more sense for the upstream NIM. IF it is a limitation the upstream NIM has I understand

Reply:
The way the multi-process stage works, we have a 1 to 1 mapping from progress engines within a stage to the worker in the process pool. This allows us to preserve completion order per-progress engine (at least that's how I remember it). We may want to remove the configuration parameter entirely.